content/docs/core/configuration.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Configuration
+description: This page discusses configuration settings for Pomerium Core.
+keywords: [core configuration]
+sidebar_label: Configuration
+---
+
+# Configuration
+
+You can configure Pomerium using either a configuration file or [environment variables](https://en.wikipedia.org/wiki/Environment_variable). If using a configuration file, the following languages are supported:
+
+- [YAML](https://yaml.org/)
+- [JSON](https://www.json.org/json-en.html)
+- [TOML](https://toml.io/en/)
+
+(Our documentation always assumes a YAML configuration file.)
+
+You can use both environment variables and a configuration file. If a particular option is set using both an environment variable and a config file key, the environment variable will take precedence.
+
+:::tip
+
+Pomerium can hot-reload route configuration details, authorization policy, certificates, and other proxy settings.
+
+:::
+
+### Configuration syntax
+
+Both configuration file keys and environment variables are case sensitive.
+
+Configuration file keys are always lowercase. Environment variables are identical to configuration file keys, except they are always uppercase.
+
+See the [Reference](/docs/reference) page for a comprehensive list of Pomerium's configuration settings.
+
+## All-in-One and Split-Service modes
+
+You can configure Pomerium using either All-in-One mode or Split-Service mode.
+
+### All-in-One mode
+
+All-in-One mode means a single Pomerium process runs all of the [four logical services](/docs/internals/architecture#component-level). You can use All-in-One mode when running Pomerium:
+
+- As a single system service or container, or
+- In a distributed environment where there are multiple processes that each handle separate [Pomerium services](/docs/internals/architecture#component-level).
+
+All-in-One mode is the default configuration mode, and the easiest way to configure Pomerium.
+
+### Split-Service mode
+
+Alternately, you can create individual configuration files (or sets of environment variables) for each Pomerium service. In Split-Service mode, each configuration file (or set of environment variables) defines which service a process will run by using the [service mode](/docs/reference/service-mode) key.
```

Comment:
I think it would be good to start with a brief explanation of how split service mode differs from all-in-one mode. Probably also good to link to the Service URL Settings page here. Maybe something like this?

```suggestion
Alternatively, you can run each Pomerium service as its own process. You might want to do this in order to scale each service independently, or so that you can enforce different permissions on each service.

To configure Pomerium in this mode, you will need to create individual configuration files (or sets of environment variables) for each Pomerium service. Each configuration should set the [Service Mode](/docs/reference/service-mode) option to tell Pomerium which service to run. You will also need the [Service URL Settings](/docs/reference/service-urls) to tell each Pomerium service how to communicate with the other Pomerium services.
```